### PR TITLE
This PR addresses [DOC-1330] Allow SEO Customization issue

### PR DESCRIPTION
**Pull request by FotieMConstant**
chore(.npmrc): add .npmrc file with auto-install-peers=true
feat(api-reference): add support for metaData configuration in ApiReference component
feat(api-reference): add @unhead/schema and unhead as dependencies in package.json

feat(ApiReference.vue): add support for metaData prop and use it to create global head instance and set SEO meta tags

chore(types.ts): import MetaFlatInput from '@unhead/schema'
feat(types.ts): add metaData property to ReferenceConfiguration type

feat(swagger-parser): add vite.config.ts file for configuring Vite build options

### CODE REVIEW

**Comment by FotieMConstant** [comment]
This PR fixes #473 issue

**Comment by FotieMConstant** [comment]
This PR fixes #470 issue

File: packages/api-reference/src/types.ts [hunk 1]
```diff
@@ -66,6 +67,8 @@ export type ReferenceConfiguration = {
   hocuspocusConfiguration?: HocuspocusConfigurationProp
   /** Key used with CNTRL/CMD to open the search modal (defaults to 'k' e.g. CMD+k) */
   searchHotKey?: string
+  /** For metadata */
```
Comment (hanspagel) [inline packages/api-reference/src/types.ts:70]:
```suggestion
  /** If used, passed data will be added to the HTML header. Read more: https://unhead.unjs.io/usage/composables/use-seo-meta */
```

**Review by hanspagel — CHANGES_REQUESTED** [review]
Awesome, thanks for taking the time! We can delete the .npmrc and I’ve added two suggestions to improve the comments. What do you think?

File: .npmrc
[the remark this answers is not in the document]
Comment (FotieMConstant) [reply .npmrc:1]: @marclave and i had some issues locally on my end yesterday during a coding session. I equally mentioned there's no need to push this file. just forgot to exclude it, my bad.

**Comment by FotieMConstant** [comment]
@hanspagel it should be all good now. Thanks for the suggestions

**Comment by marclave** [comment]
@hanspagel how are the latest changes, seems good from my review. We just gotta tidy up the merge conflicts :)

**Comment by hanspagel** [comment]
Oh, sorry. @FotieMConstant any chance you could resolve the conflicts? We’ll get this in ASAP then.

**Pull request merged**